compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.desktop.kt

```diff
@@ -785,6 +786,11 @@ internal class ComposeSceneMediator(
             })
         }
 
+        @ExperimentalComposeUiApi
+        override fun updateSelectionState(newState: TextFieldValue) {
+            textInputService.updateState(oldValue = null, newValue = newState)
```

Comment:
Diacritics is buggy on Windows in BTF2 (in BTF1 it is correct)

1. Add and switch to US/International keyboard
<img width="658" alt="image" src="https://github.com/user-attachments/assets/3b6e37ce-786a-4352-83cb-33d3e4dbfef9" />

2. Open the snippet from the PR

3. Keyboard sequence:`
```
Press Alt, s, Release Alt
Press Alt, a, Release Alt
```
Actual:
á

Expected:
ßá

https://github.com/user-attachments/assets/e6d95890-b09f-405a-b922-510d75dbc2b7

Reply:
It's not related to this PR, I've investigated and reported it [here](https://youtrack.jetbrains.com/issue/CMP-7506/Pressing-Cltr-Alt-A-on-Windows-selects-all-text-in-BasicTextField2).